lhotse/recipes/earnings21.py

```diff
@@ -0,0 +1,154 @@
+        urlretrieve_progress(
+            f"{url}", filename=zip_path, desc=f"Getting speech-datasets-main.zip"
+        )
+
+    shutil.rmtree(extracted_dir, ignore_errors=True)
+
+    with zipfile.ZipFile(zip_path) as zip:
+        for f in zip.namelist():
+            if "earnings21" in f:
+                zip.extract(f, path=target_dir)
+
+    shutil.move(target_dir / "speech-datasets-main" / "earnings21", target_dir)
+    shutil.rmtree(target_dir / "speech-datasets-main")
+
+    completed_detector.touch()
+
+    return extracted_dir
+
+
+def parse_nlp_file(filename: Pathlike):
+    with open(filename) as f:
+        transcript = list()
+        f.readline()  # skip header
+        for line in f:
+            line = line.split("|")
+            transcript.append(line[0])
+        return transcript
+
+
+def prepare_earnings21(
+    corpus_dir: Pathlike, output_dir: Optional[Pathlike] = None
+) -> Union[RecordingSet, SupervisionSet]:
+    """
+    Returns the manifests which consist of the Recordings and Supervisions.
+    When all the manifests are available in the ``output_dir``, it will simply
+    read and return them.
+
+    :param corpus_dir: Pathlike, the path of the data dir. The structure is
+        expected to mimic the structure in the github repository, notably
+        the mp3 files will be searched for in [corpus_dir]/media and transcriptions
+        in the directory [corpus_dir]/transcripts/nlp_references
+    :param output_dir: Pathlike, the path where to write the manifests.
+    :return: (recordings, supervisions) pair"""
+
+    corpus_dir = Path(corpus_dir)
+    assert corpus_dir.is_dir(), f"No such directory: {corpus_dir}"
+
+    if output_dir is not None:
+        output_dir = Path(output_dir)
+        output_dir.mkdir(parents=True, exist_ok=True)
+
+    media_dir = corpus_dir / "media"
+    audio_files = list(media_dir.glob("*.mp3"))
+    assert len(audio_files) == 44
+
+    audio_files.sort()
+    recording_set = RecordingSet.from_recordings(
+        Recording.from_file(p) for p in audio_files
+    )
+
+    nlp_dir = corpus_dir / "transcripts" / "nlp_references"
+    nlp_files = list(nlp_dir.glob("*.nlp"))
+    assert len(nlp_files) == 44
+
+    supervision_segments = list()
+    for nlp_file in nlp_files:
+        id = nlp_file.stem
+        s = SupervisionSegment(
+            id=id,
+            recording_id=id,
+            start=0.0,
+            duration=recording_set[id].duration,  # recording.duration,
+            channel=0,
+            language="English",
+            text=" ".join(parse_nlp_file(nlp_file)),
+        )
+        supervision_segments.append(s)
+    supervision_set = SupervisionSet.from_segments(supervision_segments)
+
+    validate_recordings_and_supervisions(recording_set, supervision_set)
+    if output_dir is not None:
+        supervision_set.to_json(output_dir / "supervisions.json")
+        recording_set.to_json(output_dir / "recordings.json")
```

Comment:
I recommend using `jsonl` as the default, retrospectively, I should have never added the "json" format support :)